third_party/terraform/tests/resource_app_engine_flexible_app_version_test.go

```diff
@@ -0,0 +1,126 @@
+package google
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"testing"
+)
+
+func TestAccAppEngineFlexibleAppVersion_update(t *testing.T) {
+	t.Parallel()
+
+	resourceName := fmt.Sprintf("tf-test-ae-service-%s", acctest.RandString(10))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAppEngineFlexibleAppVersionDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAppEngineFlexibleAppVersion_python(resourceName, "B2"),
+			},
+			{
+				ResourceName:            "google_app_engine_flexible_app_version.foo",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"env_variables", "deployment", "entrypoint", "service", "delete_service_on_destroy"},
+			},
+			{
+				Config: testAccAppEngineFlexibleAppVersion_python(resourceName, "B1"),
+			},
+			{
+				ResourceName:            "google_app_engine_flexible_app_version.foo",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"env_variables", "deployment", "entrypoint", "service", "delete_service_on_destroy"},
+			},
+		},
+	})
+
+}
+
+func testAccAppEngineFlexibleAppVersion_python(resourceName, instanceClass string) string {
+	return fmt.Sprintf(`
+resource "google_app_engine_flexible_app_version" "foo" {
+  version_id = "v1"
+  service    = "%s"
+  runtime    = "python"
+
+  runtime_api_version = "1"
+
+  resources {
+    cpu       = 1
+    memory_gb = 0.5
+    disk_gb   = 10
+  }
+
+  entrypoint {
+    shell = "gunicorn -b :$PORT main:app"
+  }
+
+  deployment {
+    files {
+      name = "main.py"
+      source_url = "https://storage.googleapis.com/${google_storage_bucket.bucket.name}/${google_storage_bucket_object.main.name}"
+    }
+
+    files {
+      name = "requirements.txt"
+      source_url = "https://storage.googleapis.com/${google_storage_bucket.bucket.name}/${google_storage_bucket_object.requirements.name}"
+    }
+
+    files {
+      name = "app.yaml"
+      source_url = "https://storage.googleapis.com/${google_storage_bucket.bucket.name}/${google_storage_bucket_object.yaml.name}"
+    }
+  }
+
+  liveness_check {
+    path = ""
```

Comment:
It would be nice to have more of these fields being updated if possible. Especially anything where the empty case makes sense. ie: if it makes sense to set a liveness_check.path and then remove it later by passing an empty string it should be set in the initial create and then empty string in a second update.